app/javascript/components/books.vue

```diff
@@ -1,65 +1,77 @@
 <template lang="pug">
-.books
-  .empty(v-if='books === null')
-    .fa-solid.fa-spinner.fa-pulse
-    |
-    | ロード中
-  .books__items(v-else-if='books.length !== 0')
-    .row
-      book(
-        v-for='book in books',
-        :key='book.id',
-        :book='book',
-        :isAdmin='isAdmin',
-        :isMentor='isMentor')
-  .o-empty-message(v-else)
-    .o-empty-message__icon
-      i.fa-regular.fa-sad-tear
-    p.o-empty-message__text
-      | 登録されている本はありません
+.page-body
+  nav.page-filter.form
+    .container.is-md
+      filterDropdown(
+        label='プラクティスで絞り込む',
+        :options='practices',
+        v-model='practiceId')
+  .page-content.is-books
+    .container
+      .books
+        .empty(v-if='books === null')
+          .fa-solid.fa-spinner.fa-pulse
+          |
+          | ロード中
+        .books__items(v-else-if='books.length !== 0')
+          .row
+            book(
+              v-for='book in filteredBooks',
+              :key='book.id',
+              :book='book',
+              :isAdmin='isAdmin',
+              :isMentor='isMentor')
+        .o-empty-message(v-else)
+          .o-empty-message__icon
+            i.fa-regular.fa-sad-tear
+          p.o-empty-message__text
+            | 登録されている本はありません
 </template>
 <script>
+import Choices from 'choices.js'
 import Book from './book'
+import FilterDropdown from './filterDropdown'
 
 export default {
   name: 'Books',
   components: {
-    book: Book
+    book: Book,
+    filterDropdown: FilterDropdown
   },
   props: {
     isAdmin: { type: Boolean, required: true },
-    isMentor: { type: Boolean, required: true },
-    practiceId: { type: Number, default: null, required: false }
+    isMentor: { type: Boolean, required: true }
   },
   data() {
     return {
-      books: null
+      books: null,
+      practices: [],
+      practiceId: null
     }
   },
   computed: {
-    newParams() {
-      const params = new URL(location.href).searchParams
-      if (this.practiceId) params.set('practice_id', this.practiceId)
-      return params
-    },
-    booksAPI() {
-      const params = this.newParams
-      return `/api/books.json?${params}`
+    filteredBooks() {
+      return this.practiceId
+        ? this.books.filter((book) =>
+            book.practices.some(
+              (practice) => practice.id === Number(this.practiceId)
+            )
+          )
+        : this.books
     }
   },
   created() {
-    window.onpopstate = function () {
-      location.replace(location.href)
-    }
     this.getBooks()
+    this.getPractices()
   },
   methods: {
     token() {
       const meta = document.querySelector('meta[name="csrf-token"]')
       return meta ? meta.getAttribute('content') : ''
     },
     getBooks() {
-      fetch(this.booksAPI, {
+      const uri = 'api/books.json'
```

Comment:
```suggestion
      const uri = '/api/books.json'
```

URLは相対パスだとどこを指しているかわかりづらいのと、元のURL指定も絶対パスだったのでこちらの方がいいかな〜と思います。

Reply:
e7ac87c777d1b7ef619713267081fba1f6a15d80
こちらで修正いたしました